src/main/java/com/fasterxml/jackson/core/util/TextBuffer.java

```diff
@@ -958,7 +1079,14 @@ public char[] expandCurrentSegment(int minSize) {
      * {@link #contentsAsString}, since it's not guaranteed that resulting
      * String is cached.
      */
-    @Override public String toString() { return contentsAsString(); }
+    @Override public String toString() {
+        try {
+            return contentsAsString();
+        } catch (IllegalStateException e) {
+            // can fail if the string is too large, use the default Object `toString()` if this happens
+            return super.toString();
```

Comment:
Or return constant "ERROR" or whatever. Not 100% sure why I even implemented `toString()` here originally, to be honest. It should not be used for any production thing.

Reply:
I created https://github.com/FasterXML/jackson-core/pull/938